clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/ModifyGoogleServerGroupInstanceTemplateAtomicOperation.groovy

```diff
@@ -168,6 +168,14 @@ class ModifyGoogleServerGroupInstanceTemplateAtomicOperation extends GoogleAtomi
         clonedDescription.disks = overriddenProperties.disks
 
         clonedDescription.baseDeviceName = description.serverGroupName
+
+        def bootImage = GCEUtil.getBootImage(description,
```

Comment:
Is there a particular reason for factoring out the boot image calculation from `buildAttachedDisks()`? I didn't notice any logic changes and we've duplicated this call in a few spots. If not I think this should remain as-is.

Reply:
Thanks @jtk54  I have been thinking on this for a while and could not find a better solution. Here is the thing. The only way to check if shielded VM policies should be passed is to check guestOSFeatures on Image object (**passing shielded vm policies to non-shielded image will result in error**).

To access guestOsFeatures i need Image object and i noticed that getBootImage call in buildAttachedDisks gets me this object. Of course I can leave buildAttachedDisks as it is and separately call getBootImage but that will mean calling it twice (once as part of shielded VM decision and one as part of buildAttachedDisks call). I wanted to avoid calling it twice so did this refactoring. Maybe not the best decision though.

On the other hand calling getBootImage twice will force to make changes in the below test since it expects "images" call to be made only once. Sorry for the long comment and please let me know what is the best course of action in your opinion. 

com.netflix.spinnaker.clouddriver.google.deploy.ops.CreateGoogleInstanceAtomicOperationUnitSpec > should create instance FAILED
    junit.framework.AssertionFailedError: No call to 'images' expected at this point. Still 1 call(s) to 'instances' expected.

Reply:
I see the issue -- we actually only include a [subset](https://github.com/spinnaker/clouddriver/blob/master/clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/GCEUtil.groovy#L797) of the image's attributes and a reference in the disks built up by `buildAttachedDisks()`. Furthermore, we're using the underlying raw compute model classes so we can't simply change the representation to support what we need for shielded VMs. I'm OK with the change as you proposed, but can you drop a comment in the callsites? Thanks.

Reply:
Exactly, it just seem to make sense to inject image as parameter instead of a call within buildAttachedDisks since image can be later needed for other purposes too (just like this case with shielded VMs).

Also sorry for a naive question, but what are your referring to by "callsites"? You mean places where buildAttachedDisks is called?

Reply:
Yes.